### ISSUE TITLE
Handling intermediate pages after form submit

### ISSUE THREAD

**Issue by kaizersoje** [danieldotnl/ha-multiscrape#89]
## Version of the custom_component

6.0.0

## Configuration

```yaml
multiscrape:
  - resource: "https://www.amaysim.com.au/my-account/my-amaysim/products"
    name: Amaysim
    scan_interval: 30
    log_response: true
    method: GET
    form_submit:
      submit_once: true
      resubmit_on_error: false
      resource: "https://accounts.amaysim.com.au/identity/login"
      select: "#new_session"
      input:
        username: !secret amaysim_username
        password: !secret amaysim_password
    sensor:
      - select: "#outer_wrap > div.inner-wrap > div.page-container > div:nth-child(2) > div.row.margin-bottom > div.small-12.medium-6.columns > div > div > div:nth-child(2) > div:nth-child(2)"
        name: amaysim_remaining_data
        value_template: "{{ value }}"
```

## Describe the bug

I can see from the logs that after successfully submitting the form, it says that it is getting the data from the resource url, with a response code of 200.

I have pasted the contents from the log_response file `page_soup.txt` below
```
<html><body><p>/**/('OK')</p></body></html>
```
Below is the content from the `form_submit_response_body.txt`
```
<html><body>You are being <a href="https://accounts.amaysim.com.au/identity">redirected</a>.</body></html>
```
It seems that after submitting the form, the sensor is scraping data from the intermediate page and not from the resource url.

## Debug log

```text

2022-02-01 18:51:06 DEBUG (MainThread) [custom_components.multiscrape] Amaysim # Setting up multiscrape with config:
OrderedDict([('resource', 'https://www.amaysim.com.au/my-account/my-amaysim/products'), ('name', 'Amaysim'), ('scan_interval', datetime.timedelta(seconds=30)), ('log_response', True), ('method', 'GET'), ('form_submit', OrderedDict([('submit_once', True), ('resubmit_on_error', False), ('resource', 'https://accounts.amaysim.com.au/identity/login'), ('select', '#new_session'), ('input', OrderedDict([('username', 'xxxx'), ('password', 'yyyy')]))])), ('sensor', [OrderedDict([('select', Template("#outer_wrap > div.inner-wrap > div.page-container > div:nth-child(2) > div.row.margin-bottom > div.small-12.medium-6.columns > div > div > div:nth-child(2) > div:nth-child(2)")), ('name', 'amaysim_remaining_data'), ('value_template', Template("{{ value }}")), ('force_update', False)])]), ('parser', 'lxml'), ('timeout', 10), ('verify_ssl', True)])
2022-02-01 18:51:06 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Initializing scraper
2022-02-01 18:51:06 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Found form-submit config
2022-02-01 18:51:07 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Refresh triggered
2022-02-01 18:51:07 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Continue with form-submit
2022-02-01 18:51:07 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Requesting page with form from: https://accounts.amaysim.com.au/identity/login
2022-02-01 18:51:07 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Executing form_page-request with a GET to url: https://accounts.amaysim.com.au/identity/login.
2022-02-01 18:51:17 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Response status code received: 200
2022-02-01 18:51:17 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Response headers written to file: /config/multiscrape/amaysim/form_page_response_headers.txt
2022-02-01 18:51:17 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Response headers written to file: /config/multiscrape/amaysim/form_page_response_body.txt
2022-02-01 18:51:17 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Start trying to capture the form in the page
2022-02-01 18:51:17 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Parse HTML with BeautifulSoup parser lxml
2022-02-01 18:51:18 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # The page with the form parsed by BeautifulSoup has been written to file: /config/multiscrape/amaysim/form_page_soup.txt
2022-02-01 18:51:18 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Try to find form with selector #new_session
2022-02-01 18:51:18 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Found the form, now finding all input fields
2022-02-01 18:51:18 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Found the following fields: {'utf8': '✓', 'authenticity_token': 'abc123', 'continue': None, 'ga_client_id': '', 'username': None, 'password': ''}
2022-02-01 18:51:18 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Found form action /identity/sessions and method post
2022-02-01 18:51:18 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Determined the url to submit the form to: https://accounts.amaysim.com.au/identity/sessions
2022-02-01 18:51:18 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Merged input fields with input data in config. Result: {'utf8': '✓', 'authenticity_token': 'abc123', 'continue': None, 'ga_client_id': '', 'username': 'xxxx', 'password': 'yyyy'}
2022-02-01 18:51:18 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Going now to submit the form
2022-02-01 18:51:18 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Executing form_submit-request with a post to url: https://accounts.amaysim.com.au/identity/sessions.
2022-02-01 18:51:19 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Response status code received: 302
2022-02-01 18:51:19 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Response headers written to file: /config/multiscrape/amaysim/form_submit_response_headers.txt
2022-02-01 18:51:19 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Response headers written to file: /config/multiscrape/amaysim/form_submit_response_body.txt
2022-02-01 18:51:19 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Form seems to be submitted succesfully! Now continuing to update data for sensors
2022-02-01 18:51:19 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Updating data from https://www.amaysim.com.au/my-account/my-amaysim/products
2022-02-01 18:51:19 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Executing page-request with a get to url: https://www.amaysim.com.au/my-account/my-amaysim/products.
2022-02-01 18:51:20 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Response status code received: 200
2022-02-01 18:51:20 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Response headers written to file: /config/multiscrape/amaysim/page_response_headers.txt
2022-02-01 18:51:20 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Response headers written to file: /config/multiscrape/amaysim/page_response_body.txt
2022-02-01 18:51:20 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Data succesfully refreshed. Sensors will now start scraping to update.
2022-02-01 18:51:20 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Start loading the response in BeautifulSoup.
2022-02-01 18:51:20 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Response headers written to file: /config/multiscrape/amaysim/page_soup.txt

```

**Comment by kaizersoje** [comment]
Any advice in how to address this issue?

**Comment by etamtlosz** [comment]
I'm struggling with a similar issue. Please keep us updated if you manage to get past the redirect page.

**Comment by danieldotnl** [comment]
I don't see how I can reproduce this without credentials.. Any chance you could create a dummy account and share the credentials?

**Comment by etamtlosz** [comment]
Unfortunately I can not create NOR share my credentials, since it is my electricity provider, and I my bank details, amongst other things in my account.
It looks like multiple websites around New Zealand and Australia are using the same backend. As far as I remember there's someone else reporting, on the HA forum, a similar issue with a website from AUS.

**Comment by danieldotnl** [comment]
I understand, but then I'll close this issue until someone can help me reproduce this.

**Comment by danieldotnl** [comment]
I think it's related to this issue: https://github.com/home-assistant/core/pull/61293
I'll prepare a fix.

**Comment by danieldotnl** [comment]
Please try v6.2.1

**Comment by kaizersoje** [comment]
I tried the new version but I am getting the following error.
```
Amaysim # Exception in form-submit feature. Will continue trying to scrape target page. Client error '404 Not Found' for url 'https://accounts.amaysim.com.au/identity/login' For more information check: https://httpstatuses.com/404
```

I tried to access the site through the browser and the pages are all accessible.

**Comment by danieldotnl** [comment]
@etamtlosz  for you the latest release didn't solve anything either?

**Comment by etamtlosz** [comment]
I'm not at home right now, but I'll give it a go. I would have to configure multiscrape from 0 and go from there. I'll let you know what the results are.

**Comment by etamtlosz** [comment]
I apologize for spamming your inboxes.
I'm getting a similar error to the one kaizersoje has reported a few days ago:
`2022-05-04 11:38:53 ERROR (MainThread) [custom_components.multiscrape.http] Scraper_noname_0 # Error executing POST request to url: https://secure.powershop.co.nz/customers/226294/balance.
 Error message:
 HTTPStatusError("Client error '404 Not Found' for url 'https://secure.powershop.co.nz/customers/226294/balance'\nFor more information check: https://httpstatuses.com/404")`

**Comment by danieldotnl** [comment]
@kaizersoje could you post the full debug logging?

**Comment by danieldotnl** [comment]
Please reopen when you have new input.

**Comment by kaizersoje** [comment]
I have installed v6.3.1. Here is the logs.
```
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Found the following input fields: {'utf8': '✓', 'authenticity_token': '2wWWyid92Q', 'continue': None, 'ga_client_id': '', 'username': None, 'password': ''}
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Merged input fields with input data in config. Result: {'utf8': '✓', 'authenticity_token': '2wWWyid92Q', 'continue': None, 'ga_client_id': '', 'username': xxx@yyy.com', 'password': 'abcd1234'}
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Found form action /identity/sessions and method post
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Determined the url to submit the form to: https://accounts.amaysim.com.au/identity/login
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Submitting the form
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Executing form_submit-request with a post to url: https://accounts.amaysim.com.au/identity/login.
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Response status code received: 404
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # response_headers written to file: form_submit_response_headers.txt
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # response_body written to file: form_submit_response_body.txt
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Error executing post request to url: https://accounts.amaysim.com.au/identity/login.
Error message:
 HTTPStatusError("Client error '404 Not Found' for url 'https://accounts.amaysim.com.au/identity/login'\nFor more information check: https://httpstatuses.com/404")
2022-05-17 12:40:32 ERROR (MainThread) [custom_components.multiscrape.coordinator] Amaysim # Exception in form-submit feature. Will continue trying to scrape target page.
Client error '404 Not Found' for url 'https://accounts.amaysim.com.au/identity/login'
For more information check: https://httpstatuses.com/404
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.coordinator] Amaysim # Request data from https://www.amaysim.com.au/my-account/my-amaysim/products
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Executing page-request with a get to url: https://www.amaysim.com.au/my-account/my-amaysim/products.
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Response status code received: 200
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # response_headers written to file: page_response_headers.txt
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # response_body written to file: page_response_body.txt
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Loading the content in BeautifulSoup.
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # page_soup written to file: page_soup.txt
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.coordinator] Amaysim # Data succesfully refreshed. Sensors will now start scraping to update.
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.coordinator] Finished fetching multiscrape data in 0.515 seconds (success: True)
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.sensor] Amaysim # amaysim_remaining_data # Start scraping to update sensor
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # amaysim_remaining_data # Tag selected: None
2022-05-17 12:40:32 ERROR (MainThread) [custom_components.multiscrape.sensor] Amaysim # amaysim_remaining_data # Unable to scrape data: Could not find a tag for given selector. 
Consider using debug logging and log_response for further investigation.
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.sensor] Amaysim # amaysim_remaining_data # On-error, set value to None
2022-05-17 12:40:32 DEBUG (MainThread) [custom_components.multiscrape.entity] Amaysim # amaysim_remaining_data # Sensor updated and state written to HA
```

Let me know if you need more info.

**Comment by kaizersoje** [comment]
I have installed the version. I am not getting the previous errors. However, it seems to be scraping the intermediate page after login, which was the original issue.

```
2022-05-18 13:02:02 DEBUG (MainThread) [custom_components.multiscrape.coordinator] Amaysim # Request data from https://www.amaysim.com.au/my-account/my-amaysim/products
2022-05-18 13:02:02 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Executing page-request with a get to url: https://www.amaysim.com.au/my-account/my-amaysim/products.
2022-05-18 13:02:02 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Response status code received: 200
2022-05-18 13:02:02 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # response_headers written to file: page_response_headers.txt
2022-05-18 13:02:02 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # response_body written to file: page_response_body.txt
2022-05-18 13:02:02 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Loading the content in BeautifulSoup.
2022-05-18 13:02:02 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # page_soup written to file: page_soup.txt
2022-05-18 13:02:02 DEBUG (MainThread) [custom_components.multiscrape.coordinator] Amaysim # Data succesfully refreshed. Sensors will now start scraping to update.
2022-05-18 13:02:02 DEBUG (MainThread) [custom_components.multiscrape.coordinator] Finished fetching multiscrape data in 0.133 seconds (success: True)
2022-05-18 13:02:02 DEBUG (MainThread) [custom_components.multiscrape.sensor] Amaysim # amaysim_remaining_data # Start scraping to update sensor
2022-05-18 13:02:02 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # amaysim_remaining_data # Tag selected: None
2022-05-18 13:02:02 ERROR (MainThread) [custom_components.multiscrape.sensor] Amaysim # amaysim_remaining_data # Unable to scrape data: Could not find a tag for given selector. 
Consider using debug logging and log_response for further investigation.
2022-05-18 13:02:02 DEBUG (MainThread) [custom_components.multiscrape.sensor] Amaysim # amaysim_remaining_data # On-error, set value to None
2022-05-18 13:02:02 DEBUG (MainThread) [custom_components.multiscrape.entity] Amaysim # amaysim_remaining_data # Sensor updated and state written to HA
```
I have pasted content from `page_soup.txt`

```
<html><body><p>/**/('OK')</p></body></html>
```

**Comment by kaizersoje** [comment]
This is the actual `resource` that I am trying to scrape.
![image](https://user-images.githubusercontent.com/54566217/168950599-fd5ccc25-ecfa-4768-b96a-888575e7f498.png)

**Comment by danieldotnl** [comment]
Thanks for all the info. Happy to hear at least one issues is fixed. I would really like to see the full debug logs (including the form submit stuff), to be sure nothing goes wrong there. 
If that doesn't give any pointer, I really can't do anything without credentials.

**Comment by etamtlosz** [comment]
Do you still have `log_response: true` in your sensor yaml?

**Comment by kaizersoje** [comment]
> Do you still have `log_response: true` in your sensor yaml?

Yes I do

**Comment by danieldotnl** [comment]
@kaizersoje Thanks for the logging!
There is no logging from the form-submit feature in what you posted. I can think of two reasons:
1. You removed form-submit from your config (unlikely :))
2. The logging you posted is not from the first run after a restart. In that case, multiscrape already submitted the form and somehow thinks it was successful. In the next runs it will then not submit it again as you are already authenticated.

Maybe it was successful this time and we are now running in a CSS issue, but most likely you still have the same issue.
Long story short: I need the first logging after a restart which includes form submit (like in your original issue description above).

**Comment by kaizersoje** [comment]
@danieldotnl I have not removed the form-submit from my config.
I have pasted the log after a restart.

```
2022-05-20 22:08:40 DEBUG (MainThread) [custom_components.multiscrape] # Start loading multiscrape
2022-05-20 22:08:40 DEBUG (MainThread) [custom_components.multiscrape] # Reload service registered
2022-05-20 22:08:40 DEBUG (MainThread) [custom_components.multiscrape] # Start processing config from configuration.yaml
2022-05-20 22:08:40 DEBUG (MainThread) [custom_components.multiscrape] Amaysim # Setting up multiscrape with config:
 OrderedDict([('resource', 'https://www.amaysim.com.au/my-account/my-amaysim/products'), ('name', 'Amaysim'), ('scan_interval', datetime.timedelta(seconds=30)), ('log_response', True), ('method', 'GET'), ('form_submit', OrderedDict([('submit_once', True), ('resubmit_on_error', False), ('resource', 'https://accounts.amaysim.com.au/identity/login'), ('select', '#new_session'), ('input', OrderedDict([('username', 'xxxxxx'), ('password', 'yyyyyy')]))])), ('sensor', [OrderedDict([('select', Template("#outer_wrap > div.inner-wrap > div.page-container > div:nth-child(2) > div.row.margin-bottom > div.small-12.medium-6.columns > div > div > div:nth-child(2) > div:nth-child(2)")), ('name', 'amaysim_remaining_data'), ('value_template', Template("{{ value }}")), ('force_update', False)])]), ('verify_ssl', True), ('parser', 'lxml'), ('timeout', 10)])
2022-05-20 22:08:40 DEBUG (MainThread) [custom_components.multiscrape] Amaysim # Log responses enabled, creating logging folder: /config/multiscrape/amaysim/
2022-05-20 22:08:44 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Initializing http wrapper
2022-05-20 22:08:44 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Initializing form submitter
2022-05-20 22:08:44 DEBUG (MainThread) [custom_components.multiscrape] Amaysim # Initializing scraper
2022-05-20 22:08:44 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Initializing scraper
2022-05-20 22:08:44 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Initializing http wrapper
2022-05-20 22:08:44 DEBUG (MainThread) [custom_components.multiscrape] Amaysim # Initializing coordinator
2022-05-20 22:08:52 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Starting with form-submit
2022-05-20 22:08:52 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Requesting page with form from: https://accounts.amaysim.com.au/identity/login
2022-05-20 22:08:52 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Executing form_page-request with a GET to url: https://accounts.amaysim.com.au/identity/login.
2022-05-20 22:09:21 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Response status code received: 200

2022-05-20 22:09:21 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # response_headers written to file: form_page_response_headers.txt
2022-05-20 22:09:21 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # response_body written to file: form_page_response_body.txt
2022-05-20 22:09:21 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Parse page with form with BeautifulSoup parser lxml
2022-05-20 22:09:21 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # The page with the form parsed by BeautifulSoup has been written to file: form_page_soup.txt
2022-05-20 22:09:21 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Try to find form with selector #new_session
2022-05-20 22:09:21 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Found the form, now finding all input fields
2022-05-20 22:09:21 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Form looks like this: 
<form accept-charset="UTF-8" action="/identity/sessions" class="card-row" data-action="setGAClientID" data-tid="UA-31504290-1" id="new_session" method="post" role="form"><input name="utf8" type="hidden" value="✓"/><input name="authenticity_token" type="hidden" value="wQrqMKime5w=="/>
<input id="continue" name="continue" type="hidden"/>
<input id="ga_client_id" name="ga_client_id" type="hidden" value=""/>
<div class="form-row">
<label class="form-label" for="username">amaysim number or email</label>
<input autocapitalize="none" autocorrect="off" class="username-hint" id="username" inputmode="email" name="username" type="text"/>
</div>
<div class="form-row">
<label class="form-label" for="password">Password</label>
<input id="password" name="password" type="password" value=""/>
<a class="form-link" href="javascript:void(0)" id="forgot_password">Forgot password?</a>
</div>
<button class="arrow-next" data-disable-with="logging in" name="button" type="submit">Login</button>
</form>
2022-05-20 22:09:21 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Found the following input fields: {'utf8': '✓', 'authenticity_token': 'wQrKuinswfh7=', 'continue': None, 'ga_client_id': '', 'username': None, 'password': ''}
2022-05-20 22:09:21 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Merged input fields with input data in config. Result: {'utf8': '✓', 'authenticity_token': 'wQrQNKuinswfh7sN8Pw', 'continue': None, 'ga_client_id': '', 'username': 'xxxx', 'password': 'yyyyyyyy'}
2022-05-20 22:09:21 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Found form action /identity/sessions and method post
2022-05-20 22:09:21 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Determined the url to submit the form to: https://accounts.amaysim.com.au/identity/sessions
2022-05-20 22:09:21 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Submitting the form
2022-05-20 22:09:21 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Executing form_submit-request with a post to url: https://accounts.amaysim.com.au/identity/sessions.
2022-05-20 22:09:22 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Response status code received: 200
2022-05-20 22:09:22 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # response_headers written to file: form_submit_response_headers.txt
2022-05-20 22:09:23 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # response_body written to file: form_submit_response_body.txt
2022-05-20 22:09:23 DEBUG (MainThread) [custom_components.multiscrape.form] Amaysim # Form seems to be submitted succesfully (to be sure, use log_response and check file). Now continuing to retrieve target page.
2022-05-20 22:09:23 DEBUG (MainThread) [custom_components.multiscrape.coordinator] Amaysim # Request data from https://www.amaysim.com.au/my-account/my-amaysim/products
2022-05-20 22:09:23 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Executing page-request with a get to url: https://www.amaysim.com.au/my-account/my-amaysim/products.
2022-05-20 22:09:23 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # Response status code received: 200
2022-05-20 22:09:23 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # response_headers written to file: page_response_headers.txt
2022-05-20 22:09:23 DEBUG (MainThread) [custom_components.multiscrape.http] Amaysim # response_body written to file: page_response_body.txt
2022-05-20 22:09:23 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # Loading the content in BeautifulSoup.
2022-05-20 22:09:23 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # page_soup written to file: page_soup.txt
2022-05-20 22:09:23 DEBUG (MainThread) [custom_components.multiscrape.coordinator] Amaysim # Data succesfully refreshed. Sensors will now start scraping to update.
2022-05-20 22:09:23 DEBUG (MainThread) [custom_components.multiscrape.coordinator] Finished fetching multiscrape data in 33.745 seconds (success: True)
2022-05-20 22:10:19 DEBUG (MainThread) [custom_components.multiscrape.sensor] Amaysim # amaysim_remaining_data # Setting up sensor
2022-05-20 22:10:20 DEBUG (MainThread) [custom_components.multiscrape.sensor] Amaysim # amaysim_remaining_data # Start scraping to update sensor
2022-05-20 22:10:20 DEBUG (MainThread) [custom_components.multiscrape.scraper] Amaysim # amaysim_remaining_data # Tag selected: None
2022-05-20 22:10:20 ERROR (MainThread) [custom_components.multiscrape.sensor] Amaysim # amaysim_remaining_data # Unable to scrape data: Could not find a tag for given selector. 
Consider using debug logging and log_response for further investigation.
2022-05-20 22:10:20 DEBUG (MainThread) [custom_components.multiscrape.sensor] Amaysim # amaysim_remaining_data # On-error, set value to None
2022-05-20 22:10:20 DEBUG (MainThread) [custom_components.multiscrape.entity] Amaysim # amaysim_remaining_data # Updated sensor and attributes, now adding to HA
```
I did see the form log files in the folder but they disappear after a while.

**Comment by kaizersoje** [comment]
![image](https://user-images.githubusercontent.com/54566217/169528594-b14c462b-feee-44a2-b29e-370e58e061bc.png)

**Comment by danieldotnl** [comment]
Thanks, but I'm out of analysis options without an account :(

**Comment by danieldotnl** [comment]
@RemyvdB93 It's not related to this is issue, but it is a bit of a tricky case.
Actually the only thing you need to do is to remove the `resource` from within `form_submit` in the config and remove `submit_once` as well :)

This website is not using cookies and therefore the form needs to be submitted each time. And also for this reason, we need to scrape the response of the submit action, and we cannot do a separate request to retrieve the page to be scraped.

**Comment by RemyvdB93** [comment]
You're right. It's working now! Thanks a lot

**Comment by kaizersoje** [comment]
Hi @danieldotnl, my original issue is still not working. I removed `select`, `submit_once` and `resource` from within `form_submit` as well. It still does not work after that.
